programs/address/src/lib.rs

```diff
@@ -0,0 +1,194 @@
+use anchor_lang::prelude::*;
+use ark_ff::BigInteger256;
+use light_hasher::Poseidon;
+use light_indexed_merkle_tree::{
+    array::{IndexingArray, IndexingElement, RawIndexingElement},
+    IndexedMerkleTree,
+};
+use light_utils::be_bytes_to_bigint;
+
+use crate::errors::ErrorCode;
+
+pub mod errors;
+pub mod state;
+
+declare_id!("5QPEJ5zDsVou9FQS3KCauKswM3VwBEBu4dpL9xTqkWwN");
+
+#[constant]
+pub const NR_ADDRESSES: usize = 2;
+#[constant]
+pub const MERKLE_TREE_HEIGHT: usize = 22;
+#[constant]
+pub const MERKLE_TREE_CHANGELOG: usize = 2800;
+#[constant]
+pub const MERKLE_TREE_ROOTS: usize = 2800;
+
+#[program]
+pub mod address {
+
+    use super::*;
+
+    pub fn initialize_queue(_ctx: Context<InitializeQueue>) -> Result<()> {
+        Ok(())
+    }
+
+    pub fn initialize_merkle_tree(_ctx: Context<InitializeMerkleTree>) -> Result<()> {
+        Ok(())
+    }
+
+    pub fn insert_address(
+        ctx: Context<InsertAddress>,
+        addresses: [[u8; 32]; NR_ADDRESSES],
```

Comment:
We will want to make this dynamic, probably slice of [u8;32] is best?

Reply:
Borsh doesn't allow to use slices. We can use either `Vec` or an array with known size. Pick your poison. 💀 🍸

Reply:
For now I just went with `Vec`.